devel/spicy/reference.rst

```diff
@@ -598,10 +598,118 @@ dependent types must be exported *first* in the EVT file for this to
 work. As a result, you cannot export self-recursive unit types.
 
 As you can see in the example, unit fields are always declared as
-optional on the Zeek-side, as they may not have been set during
+:zeek:see:`&optional` on the Zeek-side, as they may not have been set during
 parsing. Unit variables are non-optional by default, unless declared
 as ``&optional`` in Spicy.
 
+.. _spicy_unit_export:
+
+Controlling created Zeek types
+""""""""""""""""""""""""""""""
+
+If needed the automatic unit type exporting described above can be customized.
+The general syntax for this is:
+
+.. code-block:: spicy-evt
+
+    export SPICY_ID [with { [SIPCY_FIELD_NAME [&log], ]... }];
+    export SPICY_ID [as ZEEK_ID [ with { [SIPCY_FIELD_NAME [&log], ]...]  }];
+    export SPICY_ID [without { [SIPCY_FIELD_NAME, ]... }];
+    export SPICY_ID [as ZEEK_ID [ without { [SIPCY_FIELD_NAME, ]...]  }];
+    export SPICY_ID &log;
+    export SPICY_ID as ZEEK_ID &log;
```

Comment:
I'd still try to compress this into a single line but may be matter of taste.

Reply:
That's what I initially tried, but it always overflowed the box in the rendered pages so it required horizontal scrolling, and also required a lot of nesting. I was unsure whether adding line breaks for readability would be confusing to readers. The full expression was

```
export SPICY_ID [&log | with { [SPICY_FIELD_NAME [&log], ]...} | without { [SPICY_FIELD_NAME, ]... } | as ZEEK_ID [&log | { [SPICY_FIELD_NAME [&log], ]...} | without { [SPICY_FIELD_NAME, ]...}]]
```